### PR TITLE
Infobox builder - editing infobox elements

### PR DESCRIPTION
**Pull request by RafLeszczynski**
https://wikia-inc.atlassian.net/browse/DAT-3714
https://wikia-inc.atlassian.net/browse/DAT-3715
https://wikia-inc.atlassian.net/browse/DAT-3716

DO NOT MERGE - requires changes on app (PR to be prepared)

### CODE REVIEW

**Comment by rogatty** [comment]
`infobox-builder-image-placeholder.png` seems to have 2px of empty space that shouldn't be there.

File: front/main/app/components/infobox-builder-edit-item.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import Ember from 'ember';
+
+export default Ember.Component.extend({
+	editPanelOption: Ember.computed('item.type', function () {
+		return `infobox-builder-edit-item-${this.get('item.type')}`;
+	}),
+
+	header: Ember.computed('item.{data.label,type,value,data.default,infoboxBuilderData.index}', function () {
```
Comment (rogatty) [inline front/main/app/components/infobox-builder-edit-item.js]: Isn't this syntax awesome?

File: front/main/app/models/infobox-builder.js [hunk 2]
```diff
@@ -1,29 +1,36 @@
...
+	 * @returns {void}
+	 */
+	init() {
+		this._super(...arguments);
+		this._itemIndex = {
+			row: 0,
+			image: 0,
+			title: 0
+		};
+		this.infoboxState = Ember.A([]);
```
Comment (rogatty) [inline front/main/app/models/infobox-builder.js]: Do we need to do this explicitly? @hakubo did some investigation recently and he claims that it's done anyway by Ember's array prototype extension.

File: front/main/app/components/infobox-builder-edit-item.js (hunk 1 above)
Comment (RafLeszczynski) [inline front/main/app/components/infobox-builder-edit-item.js]: we could work on improving readability :-)

File: front/main/app/models/infobox-builder.js [hunk 3]
```diff
@@ -38,12 +45,12 @@ const InfoboxBuilderModel = Ember.Object.extend({
...
 			},
 			source: `${itemType}${index}`,
-			type: xmlTag
+			type: itemType
 		});
 	},
 
 	/**
-	 * @returns {void}
+	 * @returns {ImageItem} added item
```
Comment (rogatty) [inline front/main/app/models/infobox-builder.js]: Currently nothing is returned from this function.

File: front/main/app/models/infobox-builder.js [hunk 4]
```diff
@@ -1,29 +1,36 @@
...
 	 */
 	addToState(object) {
 		this.get('infoboxState').pushObject(object);
+
+		return object;
 	},
 
 	/**
-	 * @returns {void}
+	 * @returns {RowItem} added item
```
Comment (rogatty) [inline front/main/app/models/infobox-builder.js]: Currently nothing is returned from this function.

File: front/main/app/models/infobox-builder.js [hunk 5]
```diff
@@ -126,8 +136,41 @@ const InfoboxBuilderModel = Ember.Object.extend({
...
+	 * @param {RowItem} item
+	 * @param {string} value
+	 * @returns {void}
+	 */
+	editRowItem(item, value) {
+		const index = this.get('infoboxState').indexOf(item);
+
+		this.set(`infoboxState.${index}.data.label`, value);
+
+		if (value.trim().length) {
```
Comment (rogatty) [inline front/main/app/models/infobox-builder.js]: Would it make sense to test if this is a string? Can there be an edge case where null or undefined is sent here?

Comment (RafLeszczynski) [inline front/main/app/models/infobox-builder.js]: no, I could make it bullet proof and add the check but on other hand I'd probably prefer to throw error rather than silently skip setting source value

**Comment by rogatty** [comment]
Minor things and it'll LGTM.
